src/client/v2/indexer/lookupAssetTransactions.ts

```diff
@@ -4,121 +4,388 @@ import IntDecoding from '../../../types/intDecoding';
+   * const assetTxns = await indexerClient
+   *        .lookupAssetTransactions(assetId)
+   *        .txType("axfer")
+   *        .do();
+   * ```
+   *
+   * @param type - one of `pay`, `keyreg`, `acfg`, `axfer`, `afrz`, `appl`
+   * @category query
+   */
   txType(type: string) {
     this.query['tx-type'] = type;
     return this;
   }
 
-  // sigtype to filter with, as string
+  /**
+   * Type of signature to filter with.
+   * - sig: Standard
+   * - msig: MultiSig
+   * - lsig: LogicSig
+   *
+   * #### Example
+   * ```typescript
+   * const assetId = 163650;
+   * const assetTxns = await indexerClient
+   *        .lookupAssetTransactions(assetId)
+   *        .sigType("lsig")
+   *        .do();
+   * ```
+   *
+   * @param type - one of `sig`, `msig`, `lsig`
+   * @category query
+   */
   sigType(type: string) {
     this.query['sig-type'] = type;
     return this;
   }
 
-  // txid to filter with, as string
+  /**
+   * Lookup the specific transaction by ID.
+   *
+   * #### Example
+   * ```typescript
+   * const txId = "MEUOC4RQJB23CQZRFRKYEI6WBO73VTTPST5A7B3S5OKBUY6LFUDA";
+   * const assetId = 163650;
+   * const assetTxns = await indexerClient
+   *        .lookupAssetTransactions(assetId)
+   *        .txid(txId)
+   *        .do();
+   * ```
+   *
+   * @param txid
+   * @category query
+   */
   txid(txid: string) {
     this.query.txid = txid;
     return this;
   }
 
-  // round to filter with, as int
+  /**
+   * Include results for the specified round.
+   *
+   * #### Example
+   * ```typescript
+   * const targetBlock = 18309917;
+   * const assetId = 163650;
+   * const assetTxns = await indexerClient
+   *        .lookupAssetTransactions(assetId)
+   *        .round(targetBlock)
+   *        .do();
+   * ```
+   *
+   * @param round
+   * @category query
+   */
   round(round: number) {
     this.query.round = round;
     return this;
   }
 
-  // min round to filter with, as int
+  /**
+   * Include results at or after the specified min-round.
+   *
+   * #### Example
+   * ```typescript
+   * const minRound = 18309917;
+   * const assetId = 163650;
+   * const assetTxns = await indexerClient
+   *        .lookupAssetTransactions(assetId)
+   *        .minRound(minRound)
+   *        .do();
+   * ```
+   *
+   * @param round
+   * @category query
+   */
   minRound(round: number) {
     this.query['min-round'] = round;
     return this;
   }
 
-  // max round to filter with, as int
+  /**
+   * Include results at or before the specified max-round.
+   *
+   * #### Example
+   * ```typescript
+   * const maxRound = 18309917;
+   * const assetId = 163650;
+   * const assetTxns = await indexerClient
+   *        .lookupAssetTransactions(assetId)
+   *        .maxRound(maxRound)
+   *        .do();
+   * ```
+   *
+   * @param round
+   * @category query
+   */
   maxRound(round: number) {
     this.query['max-round'] = round;
     return this;
   }
 
-  // asset ID to filter with, as int
-  assetID(id: number) {
```

Comment:
I removed this one because the `assetID` is already required in the constructor
let me know if you want to add it back

Reply:
Nice find, I was not aware of this. Though to maintain backwards compatibility, can you keep this, but mark it as `@deprecated` and don't document it?